packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -93,26 +97,53 @@ export class DefaultCrudRepository<T extends Entity, ID>
   ) {
     const definition = entityClass.definition;
     assert(
-      !!definition,
-      `Entity ${entityClass.name} must have valid model definition.`,
+      definition.idProperties().length > 0,
+      `Entity ${entityClass.name} must have at least one id/pk property.`,
     );
 
+    this.setupModel(entityClass);
+    this.setModelClass();
+  }
+  private setupModel(entityClass: typeof Model & {prototype: T}) {
+    const definition = entityClass.definition;
+
     assert(
-      definition.idProperties().length > 0,
-      `Entity ${entityClass.name} must have at least one id/pk property.`,
+      !!definition,
+      `Entity ${entityClass.name} must have valid model definition.`,
     );
 
-    this.setupPersistedModel(definition);
+    for (const property of Object.values(definition.properties)) {
+      if (isClass(property.type)) {
+        let nestedModel = property.type as typeof Model & {
+          prototype: T;
+        };
+        this.setupModel(nestedModel);
+      } else if (isClass(property.itemType)) {
+        let nestedModel = property.itemType as typeof Model & {
+          prototype: T;
+        };
+        this.setupModel(nestedModel);
+      }
+    }
+
+    this.attachModelToDataSource(definition);
+  }
+
+  private setModelClass() {
+    const model = this.dataSource.getModel(this.entityClass.definition.name);
+    if (model) {
+      // The backing persisted model has been already defined.
```

Comment:
What if it has not been registered? 

I think the following code may work well:

```ts
const model = this.dataSource.getModel(this.entityClass.definition.name);
assert(!!model, 'The backing persisted model should have been defined by setupModel()');
this.modelClass = model as typeof juggler.PersistedModel;
```

Even better, can we modify `setupModel` to return the PersistedModel class created? The we could change the constructor as follows:

```ts
this.modelClass = this.setupModel(entityClass);
```

I would also rename `setupModel` to something like `definePersistedModel`.